handler/core/src/main/java/io/knotx/fragments/task/factory/DefaultTaskFactory.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2019 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.fragments.task.factory;
+
+import static io.knotx.fragments.handler.api.domain.FragmentResult.ERROR_TRANSITION;
+import static io.knotx.fragments.handler.api.domain.FragmentResult.SUCCESS_TRANSITION;
+import static io.knotx.fragments.task.factory.TaskOptions.NODE_LOG_LEVEL_KEY;
+
+import io.knotx.fragments.engine.Task;
+import io.knotx.fragments.engine.graph.CompositeNode;
+import io.knotx.fragments.engine.graph.Node;
+import io.knotx.fragments.engine.graph.SingleNode;
+import io.knotx.fragments.handler.action.ActionOptions;
+import io.knotx.fragments.handler.api.Action;
+import io.knotx.fragments.handler.api.ActionFactory;
+import io.knotx.fragments.handler.api.domain.FragmentContext;
+import io.knotx.fragments.handler.api.domain.FragmentResult;
+import io.knotx.fragments.task.TaskDefinition;
+import io.knotx.fragments.task.TaskFactory;
+import io.knotx.fragments.task.exception.GraphConfigurationException;
+import io.knotx.fragments.task.options.ActionNodeConfigOptions;
+import io.knotx.fragments.task.options.GraphNodeOptions;
+import io.knotx.fragments.task.options.SubtasksNodeConfigOptions;
+import io.reactivex.Single;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.Vertx;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+public class DefaultTaskFactory implements TaskFactory {
+
+  public static final String NAME = "default";
+
+  private ActionProvider actionProvider;
+
+  public DefaultTaskFactory() {
+    actionProvider = new ActionProvider(supplyFactories());
+  }
+
+  @Override
+  public String getName() {
+    return NAME;
+  }
+
+  @Override
+  public Task newInstance(TaskDefinition taskDefinition, JsonObject taskOptions, Vertx vertx) {
+    TaskOptions options = new TaskOptions(taskOptions);
+    Map<String, ActionOptions> actionNameToOptions = options.getActions();
+    if (actionNameToOptions == null) {
```

Comment:
Maybe we can make it more functional, for example
`Optional.ofNullable(actionNameToOptions).orElseThrow(() -> new GraphConfigurationException("The 'actions' property not configured!"));`